store/internal/maps/maps.go

```diff
@@ -185,8 +186,9 @@ func HashFromMap(m map[string][]byte) []byte {
 // The keys are sorted before the proofs are computed.
 func ProofsFromMap(m map[string][]byte) ([]byte, map[string]*tmcrypto.Proof, []string) {
 	sm := newSimpleMap()
-	for k, v := range m {
-		sm.Set(k, v)
+	kL := maps.Keys(m)
+	for _, k := range kL {
+		sm.Set(k, m[k])
 	}
```

Comment:
This code isn't sorting the keys, so AFAICT doesn't produce behavior any different than ranging over the map directly?